File: Assesments por disciplina/Fundamentos de Desenvolvimento com C# (P2)/Exercicio06/Student.cs
namespace Assessment.Exercicio06;

public class Student
{
    private string _name;
    private string _registrationNumber;
    private string _courseName;
    private int _averageGrade;

    public Student(string name, string registrationNumber, string courseName, int averageGrade)
    {
        this._name = name;
        this._registrationNumber = registrationNumber;
        this._courseName = courseName;
        this._averageGrade = averageGrade;
    }

    public string GetStudentInfo()
    {
        return $"| Nome: {_name}" +
               $"\n|-------------" +
               $"\n| Número da matrícula: {_registrationNumber}" +
               $"\n| Nome do curso: {_courseName}" +
               $"\n| Média da disciplinas: {_averageGrade}";
    }

    public string VerifyApproval()
    {
        return IsApproved() ? $"O aluno está aprovado com média {_averageGrade}" : $"O aluno está reprovado. Média {_averageGrade}";
    }

    private bool IsApproved()
    {
        return _averageGrade >= 7.0;
    }
}

File: Assessments por disciplina/Fundamentos de Desenvolvimento com Java (P2)/src/Questao07/Student.java
package Questao07;

public class Student {
    private String name;
    private String registrationNumber;
    private double gradeA;
    private double gradeB;
    private double gradeC;

    public Student(String name, String registrationNumber, double gradeA, double gradeB, double gradeC) {
        this.name = name;
        this.registrationNumber = registrationNumber;
        this.gradeA = gradeA;
        this.gradeB = gradeB;
        this.gradeC = gradeC;
    }

    public String isApproved() {
        return calculateAverage() >= 7.0 ? "Aprovado" : "Reprovado";
    }

    private double calculateAverage() {
        return (gradeA + gradeB + gradeC) / 3.0;
    }
}
